pyqmc/slater.py

```diff
@@ -1,100 +1,111 @@
-    superlattice = np.dot(S, cell.lattice_vectors())
-    Rpts = get_supercell_copies(cell.lattice_vectors(), S)
-    atom = []
-    for (name, xyz) in cell._atom:
-        atom.extend([(name, xyz + R) for R in Rpts])
-    supercell = gto.Cell()
-    supercell.a = superlattice
-    supercell.atom = atom
-    supercell.ecp = cell.ecp
-    supercell.basis = cell.basis
-    supercell.exp_to_discard = cell.exp_to_discard
-    supercell.unit = "Bohr"
-    supercell.spin = cell.spin * scale
-    supercell.build()
-    supercell.original_cell = cell
-    supercell.S = S
-    supercell.scale = scale
-    return supercell
-
-
-class PySCFSlaterPBC:
+from pyqmc import pbc
+
+
+def sherman_morrison_row(e, inv, vec):
+    ratio = np.einsum("ij,ij->i", vec, inv[:, :, e])
+    tmp = np.einsum("ek,ekj->ej", vec, inv)
+    invnew = (
+        inv
+        - np.einsum("ki,kj->kij", inv[:, :, e], tmp) / ratio[:, np.newaxis, np.newaxis]
+    )
+    invnew[:, :, e] = inv[:, :, e] / ratio[:, np.newaxis]
+    return ratio, invnew
+
+
+_gldict = {"laplacian": np.s_[:1], "gradient_laplacian": np.s_[0:4]}
+
+
+def _aostack_mol(ao, gl):
+    return np.concatenate(
+        [ao[_gldict[gl]], ao[[4, 7, 9]].sum(axis=0, keepdims=True)], axis=0
+    )
+
+
+def _aostack_pbc(ao, gl):
+    return [_aostack_mol(ak, gl) for ak in ao]
+
+
+class PySCFSlater:
     """A wave function object has a state defined by a reference configuration of electrons.
     The functions recompute() and updateinternals() change the state of the object, and 
     the rest compute and return values from that state. """
 
-    def __init__(self, supercell, mf, twist=None):
+    def __init__(self, mol, mf, twist=None):
         """
         Inputs:
           supercell: object returned by get_supercell(cell, S)
           mf: scf object of primitive cell calculation. scf calculation must include k points that fold onto the gamma point of the supercell
           twist: (3,) array, twisted boundary condition in fractional coordinates, i.e. as coefficients of the reciprocal lattice vectors of the supercell. Integer values are equivalent to zero.
         """
-        for attribute in ["original_cell", "S"]:
-            if not hasattr(supercell, attribute):
-                print('Warning: supercell is missing attribute "%s"' % attribute)
-                print("setting original_cell=supercell and S=np.eye(3)")
-                supercell.original_cell = supercell
-                supercell.S = np.eye(3)
-
         self.parameters = {}
-        self.param_split = {}
         self.real_tol = 1e4
+        self._coefflookup = ("mo_coeff_alpha", "mo_coeff_beta")
+
+        if hasattr(mol, "a"):
+            self._init_pbc(mol, mf, twist)
+        else:
+            self._init_mol(mol, mf)
+        self.pbc_str = "PBC" if hasattr(mol, "a") else ""
+        self._aostack = _aostack_pbc if hasattr(mol, "a") else _aostack_mol
+
+        print("scf object is type", type(mf))

```

Comment:
Please remove stray print statements in __init__()

Reply:
done